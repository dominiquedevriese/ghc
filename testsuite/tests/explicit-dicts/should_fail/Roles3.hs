{-# LANGUAGE RoleAnnotations #-}
{-# LANGUAGE TypeApplications #-}
module Roles3 where

import GHC.Base (Ord(..))

type role Set nominal
data Set a

reverseOrd :: Ord a => Ord.Dict a
reverseOrd = undefined

f3 = (undefined :: Ord (f Int) => f Int -> Set (f Int) -> Set (f Int))
     ((reverseOrd @(Maybe Int) :: Ord (f Int)))
