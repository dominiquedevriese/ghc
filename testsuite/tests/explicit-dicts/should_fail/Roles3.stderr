
Roles3.hs:13:6: error:
    • Explicit dictionary application to: ‘Ord (f Int)’
      is not allowed because the instance incoherence check
      requires type-variable arguments
    • In the expression:
        (undefined :: Ord (f Int) => f Int -> Set (f Int) -> Set (f Int))
          ((reverseOrd @(Maybe Int) :: Ord (f Int)))
      In an equation for ‘f3’:
          f3
            = (undefined :: Ord (f Int) => f Int -> Set (f Int) -> Set (f Int))
                ((reverseOrd @(Maybe Int) :: Ord (f Int)))
